provider/pkg/provider/provider.go

```diff
@@ -2756,10 +2756,11 @@ func (k *kubeProvider) tryServerSidePatch(
 		return nil, nil, false, err
 	}
 	if se, isStatusError := err.(*errors.StatusError); isStatusError {
-		// If the resource field is immutable.
-		if se.Status().Code == http.StatusUnprocessableEntity ||
+		if se.Status().Code == http.StatusUnprocessableEntity &&
 			strings.Contains(se.ErrStatus.Message, "field is immutable") {
-			return nil, nil, false, err
+			// This error occurs if the resource field is immutable.
+			// Ignore this error since this case is handled by the replacement logic.
```

Comment:
I think it's worth documenting the mini protocol here (around L2755 say). Something like:

```go
// The following are expected ways in which the server-side apply can not work;
// we return no error, but a `false` value indicating that there's no result either.
// The caller will fall back to using the client-side diff. In the particular case of an
// update to an immutable field, it is already known from the schema when
// replacement will be necessary.
```